includes/subscriptions/class-wc-payments-subscriptions-onboarding-handler.php

```diff
@@ -0,0 +1,193 @@
+<?php
+/**
+ * Class WC_Payments_Subscriptions_Onboarding_Handler
+ *
+ * @package WooCommerce\Payments
+ */
+
+defined( 'ABSPATH' ) || exit;
+
+/**
+ * A class to handle the onboarding of subscriptions products. Created subscription products will be set to draft until
+ *   onboarding is completed.
+ */
+class WC_Payments_Subscriptions_Onboarding_Handler {
+
+	/**
+	 * Transient to store subscription publish blocked error.
+	 *
+	 * @const string
+	 */
+	const WCPAY_SUBSCRIPTION_PUBLISH_BLOCKED_NOTICE = 'wcpay_subscription_onboarding_blocked_notice';
+
+	/**
+	 * Option for holding an array of product id's to publish post onboarding.
+	 *
+	 * @const string
+	 */
+	const WCPAY_SUBSCRIPTION_AUTO_PUBLISH_PRODUCTS = 'wcpay_subscription_onboarding_products';
+
+	/**
+	 * Transient to store notice that subscriptions were auto published.
+	 *
+	 * @const string
+	 */
+	const WCPAY_SUBSCRIPTION_AUTO_PUBLISH_NOTICE = 'wcpay_subscription_onboarding_published_notice';
+
+	/**
+	 * The account service instance.
+	 *
+	 * @var WC_Payments_Account|null
+	 */
+	private static $account = null;
+
+	/**
+	 * Constructor
+	 */
+	public function __construct() {
+		add_filter( 'publish_product', [ $this, 'product_published' ], 10, 3 );
+		add_action( 'admin_notices', [ $this, 'add_notices' ] );
+		add_action( 'woocommerce_payments_account_refreshed', [ $this, 'account_data_refreshed' ] );
+	}
+
+	/**
+	 * Sets the account service instance reference on the class.
+	 *
+	 * @param WC_Payments_Account $account account service instance.
+	 */
+	public static function set_account( WC_Payments_Account $account ) {
+		self::$account = $account;
+	}
+
+	/**
+	 * Convert subscriptions to drafts when using WCPay (without subscriptions) and onboarding is not complete.
+	 *
+	 * @param int $product_id Subscriptions Product id.
+	 */
+	public function product_published( int $product_id ) {
+		// We can skip if the post is not yet marked as published.
+
+		if ( $this->is_onboarding_complete() ) {
+			return;
+		}
+
+		// If Subscriptions plugin is installed we don't need to do this check.
+		if ( class_exists( 'WC_Subscriptions' ) ) {
```

Comment:
```suggestion
		if ( $this->is_subscriptions_plugin_active() ) {
```

There's a helper function you can use for this. It's the same logic but would future proof to use the helper instead. 

To get access to the `is_subscriptions_plugin_active()` function you have to add the following: 

```
use WC_Payments_Subscriptions_Utilities;
```